### PR TITLE
better way to override default parameter value in initializations/activations/regularizers

### PR DESCRIPTION
**Pull request by wlike**
1. I don't want to use lambda function as suggested/discussed in https://groups.google.com/forum/#!topic/keras-users/kc0Jsw1vTqY and https://groups.google.com/forum/#!topic/keras-users/1RbHi3I0aGE.

It's not concise and is problematic with serialization.

2. `kwargs` parameters are not suitable when there are multiple initialization methods with the same argument name. For example, `LSTM(init='uniform', inner_init='orthogonal')`, `uniform` and `orthogonal` have the same argument named `scale`.

This modification enables us to override default parameter values in the following way:
`LSTM(init='uniform(scale=0.01)', inner_init='orthogonal(scale=0.001)')`.

### CODE REVIEW

**Comment by wlike** [comment]
@fchollet, please check the detailed result of travis-ci. 5 out of 6 is passed. The one failed is test_adadelta ut.

File: keras/utils/generic_utils.py [hunk 1]
```diff
@@ -5,15 +5,27 @@
...
 def get_from_module(identifier, module_params, module_name,
                     instantiate=False, kwargs=None):
     if isinstance(identifier, six.string_types):
-        res = module_params.get(identifier)
+        idx = identifier.find('(')
+        if -1 == idx:
+            res = module_params.get(identifier)
+        else:
+            func_name = identifier[:idx]
+            aux_kwargs = eval('dict' + identifier[idx:])
```
Comment (pechersky) [inline keras/utils/generic_utils.py]: What happens if I pass in a harmful string like `"(injection=__import__('os').listdir('.'))"`? Perhaps the identifier string should just be parsed for k,v pairs, using `set` or `setattr`.

Comment (wlike) [inline keras/utils/generic_utils.py]: Use convention, and just KISS

**Comment by wlike** [comment]
image
The above is the result of Travis-CI. The failure is caused by test_adadelta in test_optimizers.py indicated by the following:
image

When I manually run the test by `pytest test_optimizers.py` with **python 2.7**, it's correct.
image

with the modification of `pytest.ini` as follows:
image

**Comment by wlike** [comment]
When I install the related packages according to code in Travis-CI 
image,
`pytest.ini` is the same with repo (without modification), and the test result is
image

**Comment by wlike** [comment]
image

The last three commits have nothing to do with the unit-test of Adadelta. But the checks succeed.
This indicates that:
1. there is something wrong with the implementation of Adadelta
or
2. there is something wrong with the travis-ci environment, which causes the unstable result of unit-test of Adadelta.

**Comment by fchollet** [comment]
This is, like, a total hack...

> I don't want to use lambda function as suggested/discussed in 

But that, or defining a named function, is exactly what you should be doing...

**Comment by wlike** [comment]
@fchollet In order to override the default value of parameter `scale` in `uniform` for example, I have to define a lambda function with unnecessary parameter `name`, and everyone who wants to override has to do the same thing. I don't think it's a good way.

**Pull request closed without merging**